src/main/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/compute/v2/OpenStackComputePlugin.java

```diff
@@ -373,24 +377,44 @@ private ComputeInstance getInstanceFromJson(String getRawResponse) {
             }
         }
 
-        ComputeInstance computeInstance = new ComputeInstance(instanceId, openStackState, hostName, ipAddresses);
+        return new ComputeInstance(instanceId, openStackState, hostName, ipAddresses);
+    }
 
-        return computeInstance;
+    protected String doGetRequest(String endpoint, OpenStackV3User clouUser) throws FogbowException {
+        String responseStr = null;
+        try {
+            responseStr = this.client.doGetRequest(endpoint, clouUser);
+        } catch (HttpResponseException e) {
+            OpenStackHttpToFogbowExceptionMapper.map(e);
+        }
+        return responseStr;
     }
 
-    protected String getRandomUUID() {
-        return UUID.randomUUID().toString();
+    protected void doDeleteRequest(String endpoint, OpenStackV3User cloudUser) throws FogbowException {
+        try {
+            this.client.doDeleteRequest(endpoint, cloudUser);
+        } catch (HttpResponseException e) {
+            OpenStackHttpToFogbowExceptionMapper.map(e);
+        }
     }
 
-    private TreeSet<HardwareRequirements> getHardwareRequirementsList() {
+    protected TreeSet<HardwareRequirements> getHardwareRequirementsList() {
         synchronized (this.hardwareRequirementsList) {
             return new TreeSet<HardwareRequirements>(this.hardwareRequirementsList);
         }
     }
 
-    private void setHardwareRequirementsList(TreeSet<HardwareRequirements> hardwareRequirementsList) {
+    protected void setHardwareRequirementsList(TreeSet<HardwareRequirements> hardwareRequirementsList) {
         synchronized (this.hardwareRequirementsList) {
             this.hardwareRequirementsList = hardwareRequirementsList;
         }
     }
+
+    public void setClient(OpenStackHttpClient client) {
```

Comment:
are you sure these last two methods need to be public? also, please consider moving them to earlier in the class file, since they initialize objects that can be used throughout the whole class.

Reply:
Make them private and added a note meaning they are for test only.